test/e2e/chaos_test.go

```diff
@@ -240,6 +257,138 @@ func testKillAllMembers(parentCtx context.Context, client crclient.Client, clust
 	}
 }
 
+// testEtcdMemberMissing ensures that the etcd cluster can be recovered having 1 member with data corruption
+func testEtcdMemberCorruption(parentCtx context.Context, client crclient.Client, cluster *hyperv1.HostedCluster) func(t *testing.T) {
+	return func(t *testing.T) {
+		g := NewWithT(t)
+		ctx, cancel := context.WithCancel(parentCtx)
+		defer cancel()
+
+		guestNamespace := manifests.HostedControlPlaneNamespace(cluster.Namespace, cluster.Name)
+		t.Logf("Hosted control plane namespace is %s", guestNamespace)
+
+		// Find etcd pods in the control plane namespace
+		etcdSts := cpomanifests.EtcdStatefulSet(guestNamespace)
+		err := client.Get(ctx, crclient.ObjectKeyFromObject(etcdSts), etcdSts)
+		g.Expect(err).NotTo(HaveOccurred(), "failed to get etcd statefulset")
+
+		etcdPods := &corev1.PodList{}
+		err = client.List(ctx, etcdPods, &crclient.ListOptions{
+			Namespace:     manifests.HostedControlPlaneNamespace(cluster.Namespace, cluster.Name),
+			LabelSelector: labels.Set(etcdSts.Spec.Selector.MatchLabels).AsSelector(),
+		})
+		g.Expect(err).NotTo(HaveOccurred(), "failed to get etcd pods")
+
+		pod := randomPods(etcdPods.Items, 1)[0]
+		command := fmt.Sprintf("find /var/lib/data/member/wal -type f -name \"*.wal\" -print0 | shuf -z -n1 | xargs -0 rm")
+
+		t.Logf("Deleting wal file from etcd pod: %s", pod.Name)
+		cmdStdout, err := e2eutil.RunCommandInPod(ctx, client, "etcd", pod.Namespace, []string{"/bin/sh", "-c", command}, pod.Spec.Containers[0].Name, 5*time.Minute)
```

Comment:
this will run the command in a random pod, not the one you have above `pod := randomPods(etcdPods.Items, 1)[0]`
if you need to run in an exact pod, use PodExecuter directly see: https://github.com/openshift/hypershift/blob/main/test/e2e/util/util.go#L962-L992

```suggestion
		cmdStdout, err := e2eutil.RunCommandInPod(ctx, client, "etcd", pod.Namespace, []string{"/bin/sh", "-c", command}, "etcd", 5*time.Minute)
```
also please hardcode the container name as well, the reader don't know what the first container is/it might change.